File: src/server/logic/DayActions.java
package server.logic;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

public class DayActions {
    private final Map<String, String> votes = new ConcurrentHashMap<>();

    public void addVote(String voterId, String targetId) {
        votes.put(voterId, targetId);
    }

    public String getLynchedPlayer() {
        if (votes.isEmpty()) return null;
        
        Map<String, Integer> voteCounts = new ConcurrentHashMap<>();
        for (String targetId : votes.values()) {
            voteCounts.put(targetId, voteCounts.getOrDefault(targetId, 0) + 1);
        }

        String mostVoted = null;
        int maxVotes = 0;
        boolean isTie = false;

        for (Map.Entry<String, Integer> entry : voteCounts.entrySet()) {
            if (entry.getValue() > maxVotes) {
                maxVotes = entry.getValue();
                mostVoted = entry.getKey();
                isTie = false;
            } else if (entry.getValue() == maxVotes) {
                isTie = true;
            }
        }
        
        return isTie ? null : mostVoted;
    }

    public void clearVotes() {
        votes.clear();
    }
}

File: src/server/logic/DayActions.cs
using System.Collections.Concurrent;
using System.Collections.Generic;

namespace WerewolfGame.Server.Logic
{
    public class DayActions
    {
        private readonly ConcurrentDictionary<string, string> votes = new ConcurrentDictionary<string, string>();

        public void AddVote(string voterId, string targetId)
        {
            votes[voterId] = targetId;
        }

        public string? GetLynchedPlayer()
        {
            if (votes.IsEmpty) return null;

            var voteCounts = new ConcurrentDictionary<string, int>();
            foreach (var targetId in votes.Values)
            {
                voteCounts.AddOrUpdate(targetId, 1, (key, oldValue) => oldValue + 1);
            }

            string? mostVoted = null;
            int maxVotes = 0;
            bool isTie = false;

            foreach (var entry in voteCounts)
            {
                if (entry.Value > maxVotes)
                {
                    maxVotes = entry.Value;
                    mostVoted = entry.Key;
                    isTie = false;
                }
                else if (entry.Value == maxVotes)
                {
                    isTie = true;
                }
            }

            return isTie ? null : mostVoted;
        }

        public void ClearVotes()
        {
            votes.Clear();
        }
    }
}